src/validators/dataclass.rs

```diff
@@ -474,11 +474,7 @@ impl BuildValidator for DataclassValidator {
             None
         };
 
-        let fields = schema
-            .get_as_req::<&PyList>(intern!(py, "fields"))?
-            .iter()
-            .map(|s| Ok(s.downcast::<PyString>()?.into_py(py)))
-            .collect::<PyResult<Vec<_>>>()?;
```

Comment:
Why don't we need this anymore?

Reply:
Type inference is able to work out that we want `fields` to be `Vec<Py<PyString>>`, so best to let PyO3 decide the most efficient way to achieve that.

This code was flagged to me because `&PyList` is now removed as a type (we always wrap it as `Py<PyList>` or `Bound<'_, PyList>` now), so I simplified :)

Reply:
Great!